provider/metrics/cache.go

```diff
@@ -0,0 +1,37 @@
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+	stats "github.com/virtual-kubelet/virtual-kubelet/node/api/statsv1alpha1"
+	v1 "k8s.io/api/core/v1"
+)
+
+func WrapCachedPodStatsGetter(ttlSeconds int, getter podStatsGetter) *cachePodStatsGetter {
+	return &cachePodStatsGetter{
+		wrappedGetter: getter,
+		cache:         cache.New(time.Duration(ttlSeconds)*time.Second, 10*time.Minute),
+	}
+}
+
+//Adding cache capability into podStatsGetter
+type cachePodStatsGetter struct {
+	wrappedGetter podStatsGetter
+	cache         *cache.Cache
```

Comment:
There are two scenarios to use cache in this metrics:
1. first is here. cache the result of getPodStats to avoid frequent ACI's API calls. 
2. second is to cache the cumulative value. The new cumulative value should be the old one + delta value.

Both of scenarios just need to:
1. cache the latest value.
2. automatically clean the old data if cache not be used for a while because corresponding Pod has been deleted, or refresh the cumulative data if it not update for a while. 

The [patrickmn/go-cache](https://github.com/patrickmn/go-cache) is what I need. Its codes is straightforward and have the only features I need.   I didn't know the golang-lru library. It look work for first scenario as well. But there is small problem the second scenario. We don't need LRU algorithm. I hope the cache data always there, otherwise the cumulative value will be reset.